src/main/java/org/esbtools/auth/spring/CertEnvironmentVerificationFilter.java

```diff
@@ -0,0 +1,80 @@
+package org.esbtools.auth.spring;
+
+import java.io.IOException;
+import java.security.cert.X509Certificate;
+
+import javax.naming.NamingException;
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.esbtools.auth.util.EnvironmentUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.web.WebAttributes;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+public class CertEnvironmentVerificationFilter extends OncePerRequestFilter {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(CertEnvironmentVerificationFilter.class);
+
+  private final EnvironmentUtils envUtils;
+
+  public CertEnvironmentVerificationFilter(String environment) {
+    envUtils = (null == environment) ? null : new EnvironmentUtils(environment);
+
+    LOGGER.info("Cert Environment: " + ((environment == null) ? "Not Set" : environment));
+  }
+
+  @Override
+  public void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain chain) throws IOException, ServletException {
+    if (null != envUtils) {
```

Comment:
You also want to check if what is configured is not an empty string (i.e. "").  In the past an empty environment signified you could skip environment validation.  I think that the way this is currently written, an empty environment config would still be validated (because it is not null).  

We could also choose to interpret null for the environment configuration as "doesn't need to be validated", we just need to be consistent and allow for that in CertLoginLdapLoginModule (currently requires environment to be present in config) and whatever else uses this.

We could instead a method to the Environment class called requiresValidation() that returns false when the value is null or empty string, and true otherwise (I think I like this option the best).  We could also check the allAccessOu value in that method too since that is another factor in determining whether or not to do validation logic.

Reply:
> You also want to check if what is configured is not an empty string (i.e. "")

Do we need to handle blank spaces also? eg. " "

> We could also choose to interpret null for the environment configuration as "doesn't need to be validated"

That is exactly what CertEnvironmentVerificationFilter does, it just handles it at that layer instead of in Environment. I did it at that level because I didn't have enough context to make changes to the logic flow of CertLdapLoginModule.

If we want this to be default behaviour, then it can definitely be moved to the Environment / requiresValidation.